src/App.tsx

```diff
@@ -1,26 +1,312 @@
+  }
+
+  function deleteTodo(todoId: number) {
+    setLoadingTodoId(todoId);
+    deleteTodos(todoId)
+      .then(() => {
+        setTodos(currentTodos =>
+          currentTodos.filter(todo => todo.id !== todoId),
+        );
+        setCounterOfActiveTodos(currentCount => currentCount - 1); // Уменьшаем счетчик
+      })
+      .catch(err => setError('Unable to delete a todo' + err.message))
+      .finally(() => setLoadingTodoId(null));
+  }
+
+  function deleteCompletedTodos() {
+    setLoading(true);
+    getTodos()
+      .then(fetchedTodos => {
+        const completedTodos = fetchedTodos.filter(todo => todo.completed);
+
+        const deletionPromises = completedTodos.map(todo =>
+          deleteTodos(todo.id)
+            .then(() => {
+              // Уменьшаем счетчик после успешного удаления
+              setCounterOfActiveTodos(currentCount => currentCount - 1);
+            })
+            .catch(err => {
+              // Обрабатываем ошибку, если задача не была удалена
+              setError('Unable to delete a todo' + err.message);
+            }),
+        );
+
+        return Promise.all(deletionPromises);
+      })
+      .then(() => {
+        // Обновляем состояние todos, удаляя успешно удаленные задачи
+        setTodos(currentTodos => currentTodos.filter(todo => !todo.completed));
+      })
+      .finally(() => {
+        setLoading(false); // Optional: Hide loading state
+      });
+  }
+
+  function updateTodoStatus(todoId: number, completed: boolean) {
+    const todoToUpdate = todos.find(todo => todo.id === todoId);
+
+    if (!todoToUpdate) {
+      return;
+    }
+
+    setLoadingTodoId(todoId);
+    updateTodos({ ...todoToUpdate, completed })
+      .then(updatedTodo => {
+        setTodos(currentTodos =>
+          currentTodos.map(todo =>
+            todo.id === updatedTodo.id ? updatedTodo : todo,
+          ),
+        );
+      })
+      .catch(err => setError('Unable to delete a todo' + err.message))
+      .finally(() => setLoadingTodoId(null));
+  }
+
+  const handleFilterChange = (newFilter: Filter) => setFilter(newFilter);
+
+  const filteredTodos = todos.filter(todo => {
+    switch (filter) {
+      case Filter.ACTIVE:
+        return !todo.completed;
+      case Filter.COMPLETED:
+        return todo.completed;
+      default:
+        return true;
+    }
+  });
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">
-          React Todo App - Load Todos
-        </a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
```

Comment:
Consider creating separate components for header, todolist and footer

Reply:
<img width="650" alt="image" src="https://github.com/user-attachments/assets/940cbe0c-b8a3-478c-a7f7-12e95af3a1e5">
better to implement it